drivers/spi/spi_sc18is606.c

```diff
@@ -0,0 +1,375 @@
+/*
+ * Copyright (c) 2025, tinyvision.ai
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_sc18is606_spi
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(spi_sc18is606, CONFIG_SPI_LOG_LEVEL);
+
+#include <zephyr/kernel.h>
+#include <stdint.h>
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/sys/util.h>
+#include "spi_context.h"
+
+#define SC18IS606_CONFIG_SPI 0xF0
+#define CLEAR_INTERRUPT      0xF1
+#define IDLE_MODE            0xF2
+#define SC18IS606_LSB_MASK   GENMASK(5, 5)
+#define SC18IS606_MODE_MASK  GENMASK(3, 2)
+#define SC18IS606_FREQ_MASK  GENMASK(1, 0)
+
+#include "spi_sc18is606.h"
+
+struct nxp_sc18is606_data {
+	struct k_mutex bridge_lock;
+	struct spi_context ctx;
+	uint32_t frequency;
```

Comment:
```suggestion
	uint8_t frequency_idx;
```
for usage clarity in code :)